force-app/main/default/classes/AccountTriggerTest.cls

```diff
@@ -0,0 +1,245 @@
+@isTest
+public class AccountTriggerTest {
+  //more than 40,000 accounts have to be updated regularly
+  private static final Integer NUM_OF_ACCOUNTS = 4;
+  private static final Integer NUM_OF_CHILD_ACCOUNTS = 10;
+  private static final Decimal DISCOUNT = 50;
+
+  @TestSetup
+  static void makeData() {
+    List<Account> parentAccounts = new List<Account>();
+
+    for (Integer i = 0; i < NUM_OF_ACCOUNTS; i++) {
+      Account accParent = new Account(
+        Name = 'Test Parent Account ' + i,
+        Discount__c = 20
+      );
+      parentAccounts.add(accParent);
+    }
+    Test.startTest();
+    insert parentAccounts;
+    Test.stopTest();
+  }
+
+  @isTest
+  private static void updateDiscountOnParent_unlockedDiscount() {
```

Comment:
ideally, a test method name expresses what s being accomplished in a business understandable language
a pattern I recently learned about is
itShouldUpdateDiscountOnUnlockedChildrenIfParentDiscountChanges

The same goes for assert messages. Treat them as a way of documenting intent. 

Also variable names: how about parentDiscount to make it more clear, which discount we are talking about.

Reply:
you can use DISCOUNT, I just say it might make sense to rename it locally
`expectedLockedChildDiscount = DISCOUNT`